src/main/java/com/nashtech/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,44 @@
+package com.nashtech.exception;
+
+
+import org.apache.kafka.common.KafkaException;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.reactive.function.client.WebClientException;
+
+/**
+ * Global exception handler for the application.
+ */
+
+@ControllerAdvice
+public class GlobalExceptionHandler extends RuntimeException {
+	
+	
+	/**
+ 	  * Exception handler method to handle {@link KafkaException}.
+ 	  * This method handles any exceptions of type {@link KafkaException} that might occur
+ 	  * during message sending to Kafka and returns an HTTP response with status code 400 (Bad Request).
+ 	  *
+          * @param kafkaException The {@link KafkaException} that occurred during message sending.
+	  * @return An HTTP response entity with status code 400 (Bad Request).
+ 	*/
+        @ExceptionHandler(value = KafkaException.class)
+        public ResponseEntity<Object> KafkaExceptionHandler(KafkaException kafkaException) {
```

Comment:
Please add the error message in the response as to why we received Kafka or Webclient execption.